Source/WarpX.H

```diff
@@ -1271,6 +1271,38 @@ public:
     */
     void ComputeEightWaysExtensions();
 
+    AMREX_GPU_DEVICE
+    static int ComputeNBorrowOneFaceExtension(const amrex::Dim3 cell, const amrex::Real S_ext,
+                                              const amrex::Array4<amrex::Real>& S_red,
+                                              const amrex::Array4<int>& flag_info_face,
+                                              const amrex::Array4<int>& flag_ext_face, const int idim);
+
+    AMREX_GPU_DEVICE
+    static int ComputeNBorrowEightFacesExtension(const amrex::Dim3 cell, const amrex::Real S_ext,
+                                                 const amrex::Array4<amrex::Real>& S_red,
+                                                 const amrex::Array4<amrex::Real>& S,
+                                                 const amrex::Array4<int>& flag_info_face,
+                                                 const int idim);
+
+    template <typename T>
```

Comment:
Could you please add comments explaining what each of these functions does?

Reply:
Also, for these template functions, it doesn't make sense to declare them here, and then define them in a single .cpp file. If they're meant to be usable by code in other files, the definition must appear here in the header. If they're not meant to be generally usable, then they don't need to be declared here at all. They could just be free functions in the .cpp file.

Reply:
Thanks, I should have addressed all your comments.